test/unit/buffer_spec.lua

```diff
@@ -293,11 +298,131 @@ describe('buffer functions', function()
     it('should print the number of lines in the buffer', function()
       buffer.setfname(globals.curbuf, to_cstr("test/unit/buffer_spec.lua"), NULL, 1)
 
-      local width = build_stl_str_hl("%L")
+      local width = build_stl_str_hl("%L", 80)
 
       eq(1, width)
       eq("1", helpers.ffi.string(output_buffer, width))
 
     end)
+
+    it('should right align when using =', function()
+      buffer.setfname(globals.curbuf, to_cstr("test/unit/buffer_spec.lua"), NULL, 1)
+
+      local expected_width = 20
+      local width = build_stl_str_hl("neo%=vim", expected_width)
+
+      eq(width, expected_width)
+      eq("neo              vim", get_str(output_buffer, expected_width))
+    end)
+
+    it('should approximately center text when using %=text%=', function()
```

Comment:
Based on the fact that tests are nearly identical I would suggest to write all of them like this:

```lua
    local peqtest = function(name, width, stl, expected_stl)
      it(name, function()
        buffer.setfname(globals.curbuf, to_cstr("test/unit/buffer_spec.lua"), NULL, 1)

        local expected_width = width
        local actual_width = build_stl_str_hl(stl, expected_width)

        eq(actual_width, expected_width)
        eq(expected_stl, get_str(output_buffer, expected_width))
      end)
    end

    peqtest('should approximately center text when using %=text%=', 20, 'abc%=neovim%=def',
            'abc    neovim    def')
```

: i.e. use function to generate tests in place of copy-pasting test function over and over.